cmd/kcp-front-proxy/options/authentication.go

```diff
@@ -49,12 +56,45 @@ func (c *Authentication) ApplyTo(authenticationInfo *genericapiserver.Authentica
 		}
 		authenticationInfo.Authenticator = x509.NewDynamic(clientCAProvider.VerifyOptions, x509.CommonNameUserConversion)
 	}
+
+	// only pass on those groups to the shards we want
+	if len(c.AllowedGroups) > 0 || len(c.DisallowedGroups) > 0 {
+		filter := &kcpauthentication.GroupFilter{
+			Authenticator:    authenticationInfo.Authenticator,
+			AllowedGroups:    sets.NewString(),
+			DisallowedGroups: sets.NewString(),
+		}
+		authenticationInfo.Authenticator = filter
+
+		for _, g := range c.AllowedGroups {
+			if strings.HasSuffix(g, "*") {
```

Comment:
can't we use https://pkg.go.dev/path/filepath#Glob ?

Reply:
We could. But then I would rather choose regexps than non-standard golang globbing. Also note that Glob has special meaning for slashes.